File: Examples/ClassDiagram/Dart/library.dart
import 'playlist.dart';

/// The user's whole collection: every playlist in one place.
class Library {
  Library([List<Playlist>? playlists]) : playlists = playlists ?? <Playlist>[];

  final List<Playlist> playlists;

  void addPlaylist(Playlist playlist) {
    playlists.add(playlist);
  }

  Playlist? playlist(String named) {
    for (final playlist in playlists) {
      if (playlist.name == named) return playlist;
    }
    return null;
  }
}

File: Examples/ClassDiagram/TypeScript/Library.ts
import { Playlist } from "./Playlist";

/** The user's whole collection: every playlist in one place. */
export class Library {
    constructor(readonly playlists: Playlist[] = []) {}

    addPlaylist(playlist: Playlist): void {
        this.playlists.push(playlist);
    }

    playlist(named: string): Playlist | undefined {
        return this.playlists.find((p) => p.name === named);
    }
}
